src/components/NewExperiment.vue

```diff
@@ -1198,22 +1274,91 @@ export default Vue.extend({
           });
       });
 
-      // parsedRows = [[["name", "a"], ["measurement", 5], ["uncertainty", null]]]
-      parsedRows.forEach((rowPairs, rowIx) => {
-        if (!table.items[rowOffset + rowIx]) {
-          // Create excess rows.
-          table.items.push({ temporaryId: uuidv4() });
-        }
+      let itemType;
+      let items;
+      if (table.name === "Samples") {
+        itemType = "condition";
+        items = this.tables.conditions.items.filter(
+          condition => condition.name
+        );
+      } else {
+        itemType = "sample";
+        items = this.tables.samples.items.filter(sample => sample.name);
+      }
+
+      // Ask which condition/sample pasted data belongs to
+      const selection =
+        table.name === "Conditions"
+          ? Promise.resolve(true)
+          : this.$promisedDialog(SelectDialog, {
+              itemType: itemType,
+              items: items
+            });
```

Comment:
Here's how I would do it, but I don't have a strong preference ;
It is convoluted anyway
```ts
      // Ask which condition/sample pasted data belongs to
      let dialogSelection;
      if (table.name === "Samples") {
        dialogSelection = this.$promisedDialog(SelectDialog, {
          itemType: "condition",
          items: this.tables.conditions.items.filter(({ name }) => name)
        }).then(selected => (selected ? [["condition", selected]] : []));
      } else if (table.name === "Conditions") {
        dialogSelection = Promise.resolve([]);
      } else {
        dialogSelection = this.$promisedDialog(SelectDialog, {
          itemType: "sample",
          items: this.tables.samples.items.filter(({ name }) => name)
        }).then(selected => (selected ? [["sample", selected]] : []));
      }

      dialogSelection.then(rowPairsFromDialog => {
         ...
          [...rowPairs, ...rowPairsFromDialog].forEach(([property, value]) => {
            Vue.set(table.items[rowOffset + rowIx], property, value);
          });
      })
```

Reply:
👍